pytorch_lightning/accelerators/ddp_accelerator.py

```diff
@@ -188,6 +189,13 @@ def get_device_ids(self):
         device_ids = [self.trainer.root_gpu]
         return device_ids
 
+    def sync_processes_best_model_path(self):
+        best_model_path = self.broadcast(self.trainer.checkpoint_callback.best_model_path)
```

Comment:
i'm curious why the broadcast is needed. the issue in https://github.com/PyTorchLightning/pytorch-lightning/pull/4323 seems like metrics aren't synced globally, leading to inconsistent paths across ranks. 

if we want to cover this case, do we also need to broadcast other properties like `best_model_score` from rank 0?

Reply:
I don't think #4323 is really covered well in this PR, and I was questioning whether whether to include the broadcast or not... the primary thing we have to do is await all processes completion of saving for #4319. Forcing the same best model path on all ranks allows them to correctly pick the best model, as rank 0 is the only rank that has control over the saving of checkpoints.

If you're getting different scores across processes, this is a deeper issue of how to consolidate them. I think moving forward we should push the metrics API which handles syncing of metrics across ranks: https://pytorch-lightning.readthedocs.io/en/latest/metrics.html

I'm still curious if theres a better way to handle the natively the different val scores, but unsure if it's required for this PR. Maybe omitting the broadcast and just doing a barrier would suffice, thoughts @ananthsub?

Reply:
> If you're getting different scores across processes, this is a deeper issue of how to consolidate them. I think moving forward we should push the metrics API which handles syncing of metrics across ranks: https://pytorch-lightning.readthedocs.io/en/latest/metrics.html

I totally agree

I'm fine with the barrier here. If we really need the broadcasts, can they move to the model checkpoint callback itself? the callback could implement the hook for `on_fit_end` and handle it there so we can keep the properties in sync in a single location

Reply:
yeah good shout, will do!

Reply:
@ananthsub let me know if you agree with the changes I just made :)

Reply:
Looks good! I'm debugging what I think is the same race condition in DDP torchelastic. I think we need the same barrier across the DDP accelerators

Reply:
Want me to add them across the accelerators in the same place?

Reply:
@williamFalcon is this needed across all DDP accelerators? ddp/ddp_cpu/ddp2 + slurm and torchelastic variants?

Reply:
yeah. i think any dist type of call needs to be implemented in each accelerator with however they do it and then call it from the accelerator.

Reply:
ie: change

if tpu:
  x()

if gpu:
  y()

...

TO

TPU:

def shared_fx():
   x()

GPU

def shared_fx():
   y()

then you use
accelerator.shared_fx()